### DISCUSSION TITLE
[keyball61]トラックボールのマウス移動時にチャタリングが発生する

### DISCUSSION THREAD

**Discussion by sota1235** [Yowkees/keyball#423] [Q&A]
keyball61を組み立て後、2ヶ月ほど問題なく愛用していたのですが自宅で持ち歩いた後チャタリングが発生するようになってしまいました。
症状としては下記と同じで、移動時に127, -127の入力が突然入ります。
https://github.com/Yowkees/keyball/discussions/289

過去のQ&Aを見たところ

- L字7PコンスルーのU字型のピンをピンセット等で広げる
- ProMicroの不良を疑い、交換する

あたりが対応になるのかなと思ってるんですが他に何か問題を切り分ける方法はありますでしょうか…？

なお、1点目のL字7Pコンスルーについてはピンセットで広げようとしたところ破損してしまったので先日、部品を別途購入した上で交換済みです。
新品のものをはんだ付けして症状が変わらないのでL字7Pコンスルーの接触不良の可能性は低いのかなと考えていますがこの考えが正しいかもご教授いただけると… :bow:

**Comment by koron** [comment]
疑わしい順に上げていきます。

1. ボールや支持球やレンズの間のゴミや汚れ、またはそれに伴うボールとレンズ間の距離の変化
2. 光学センサーの故障
3. Pro Microの電圧不良&ハンダ不良によるクロストーク

より詳しく検証するために、以下の写真を撮って共有していただけないでしょうか。

1. ボール
2. ボールを外してケース内(レンズと支持球が確認できるように
3. 本体の裏面(ハンダ付けした面)全体が映るように
4. Pro Microを外して裏表両面

**Comment by sota1235** [comment]
回答ありがとうございます。

以下がトラックボールの写真です(表裏)。

![PXL_20231119_115250069 MP](https://github.com/Yowkees/keyball/assets/2968860/2d6c673b-0453-499a-b8f8-b61b1f8ab7ce)
![PXL_20231119_115254740 MP](https://github.com/Yowkees/keyball/assets/2968860/196c3b87-5516-48b2-a60d-857fc1264575)

こちらがボールを外したケース内の写真です。

![PXL_20231119_115300952 MP](https://github.com/Yowkees/keyball/assets/2968860/b325de34-f1e3-40a8-bfc2-edb7fe135449)
![PXL_20231119_115307599 MP](https://github.com/Yowkees/keyball/assets/2968860/94231c14-45a4-469f-ac1e-2cda2299bcfb)

こちらが本体の裏面です。

![PXL_20231119_115753837 MP](https://github.com/Yowkees/keyball/assets/2968860/3fa93e63-4ea9-4daa-bf5d-ff61296e5634)
![PXL_20231119_115802215 MP](https://github.com/Yowkees/keyball/assets/2968860/3b7e624c-bc9d-4e28-94d0-7df6e3d77a1a)

`4. Pro Microを外して裏表両面` に関してはまだ作業時間が取れておらず追って共有させてください 🙇 (ハンダ除去の作業に慣れてないので時間がかかりそうです)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。またkoron様もアドバイスありがとうございます。

現在までにご共有いただいた半田付けは特段問題はなさそうですが、7Pコンスルー近くのジャンパーにクラック（ヒビ）が入って導通不良になる事がありますので再加熱してみても良いかもしれません。

ProMicroはピンソケットとピンヘッダーで固定されているので引き抜くだけですが、取れないでしょうか？

また、同症状でトラックボール基板の半田付け不良が原因の方も多いのでトラックボール基板の写真も見せていただければ幸いです。

以上、お手数ですがご対応お願いいたします。

**Comment by sota1235** [comment]
ご回答ありがとうございます！

> 現在までにご共有いただいた半田付けは特段問題はなさそうですが、7Pコンスルー近くのジャンパーにクラック（ヒビ）が入って導通不良になる事がありますので再加熱してみても良いかもしれません。

こちら試してみたのですが症状変わらずでした…

> ProMicroはピンソケットとピンヘッダーで固定されているので引き抜くだけですが、取れないでしょうか？

ちょっと試してみたのですがかなり固くて取り外しが難しい気がしてます。もう少し試してみます…！

> また、同症状でトラックボール基板の半田付け不良が原因の方も多いのでトラックボール基板の写真も見せていただければ幸いです。

こちら、表裏の写真を撮りました。
一応ですが基盤は自分で組み立てた際にははんだ付けをミスしたので組み立て済みのものを白金工房さんで購入させていただきました。(私が作業した場合はおかしい可能性が高い気がするので参考までに…！)

![PXL_20231123_135359086 MP](https://github.com/Yowkees/keyball/assets/2968860/787bd09a-e0e0-4345-8355-059ebd86b97d)
![PXL_20231123_135406952 MP](https://github.com/Yowkees/keyball/assets/2968860/4c975be5-7557-4fb5-b90f-2157339fbe5b)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

半田付け済のトラックボール基板をご使用でしたら、動作確認済のものをお送りしていますので、トラボ基板には問題ないと思います。
ProMicroが引き抜きできないということで、もしかするとピンソケット内に半田が流れ込み、となりとショートしている個所があるかも知れません。
どうしても取り外せないようでしたら、お手数ですが白銀ラボウェブサイトよりご連絡ください。郵送による修理等をご提案させていただきます。

以上、お手数ですがご対応お願いいたします。

**Comment by sota1235** [comment]
返信遅れ申し訳ありません。

> ProMicroが引き抜きできないということで、もしかするとピンソケット内に半田が流れ込み、となりとショートしている個所があるかも知れません。
> どうしても取り外せないようでしたら、お手数ですが白銀ラボウェブサイトよりご連絡ください。郵送による修理等をご提案させていただきます。

こちら、もう少し試してみます。

あと、アップデートなんですがチャタリング自体は特になんのきっかけもなく直ってしまいました…。
どこかの接触不良だったのかもしれません。

ただ別問題として右下の2キー(トラックボールの右の2つのキー)をクリックするとたまにmouseを触ってる判定になってしまい、automatic mouse layerが誤作動するようになりました。
こちらに関してはfirmwareの焼き直し等を試そうと思ってますがもしかしたら別QAとして相談させてもらうかもしれません。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。
トラックボール基板の接続が、少々不安定なのかも知れません。
7Pコンスルーのピン変形や、トラボ基板が奥まで差さっているかご確認ください。
トラックボールケースに触ったときに不安定になっているかもです。

お手数ですがご確認お願いいたします。
